src/Features/VisualBasic/Portable/ConvertAutoPropertyToFullProperty/VisualBasicConvertAutoPropertyToFullProperty.vb

```diff
@@ -0,0 +1,105 @@
+﻿' Copyright(c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt In the project root For license information
+
+Imports System.Composition
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.CodeRefactorings
+Imports Microsoft.CodeAnalysis.ConvertAutoPropertyToFullProperty
+Imports Microsoft.CodeAnalysis.Editing
+Imports Microsoft.CodeAnalysis.Options
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.ConvertAutoPropertyToFullProperty
+    <ExportCodeRefactoringProvider(LanguageNames.VisualBasic, Name:=NameOf(VisualBasicConvertAutoPropertyToFullPropertyCodeRefactoringProvider)), [Shared]>
+    Friend Class VisualBasicConvertAutoPropertyToFullPropertyCodeRefactoringProvider
+        Inherits AbstractConvertAutoPropertyToFullPropertyCodeRefactoringProvider
+
+        Private Const Underscore As String = "_"
+
+        Friend Overrides Function GetProperty(token As SyntaxToken) As SyntaxNode
+            Dim containingProperty = token.Parent.FirstAncestorOrSelf(Of PropertyStatementSyntax)
+            If containingProperty Is Nothing Then
+                Return Nothing
+            End If
+
+            Dim start = If(containingProperty.AttributeLists.Count > 0,
+                containingProperty.AttributeLists.Last().GetLastToken().GetNextToken().SpanStart,
+                containingProperty.SpanStart)
+
+            ' Offer this refactoring anywhere in the signature of the property.
+            Dim position = token.SpanStart
+            If position < start Then
+                Return Nothing
+            End If
+
+            If containingProperty.HasReturnType() AndAlso
+                position > containingProperty.GetReturnType().Span.End Then
+                Return Nothing
+            End If
+
+            Return containingProperty
+        End Function
+
+        ''' <summary>
+        ''' In VB, auto properties have an implicit backing field that is named using the property 
+        ''' name preceded by an underscore. We will use this as the field name so we don't mess up 
+        ''' any existing references to this field.
+        ''' </summary>
+        Friend Overrides Function GetFieldNameAsync(document As Document, propertySymbol As IPropertySymbol, cancellationToken As CancellationToken) As Task(Of String)
+            Return Task.FromResult(Underscore + propertySymbol.Name)
```

Comment:
Two things.  You can just use "_" here instead of having the constant.  Constnats are good when the name tells you something interesting about hte value.  like "Second_in_a_year = 31_536_000".  They're also good if you're going to use the value several times.  When you only use it once, and the name of the variable is exactly the same as the value, they're not needed.